scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -2191,13 +2272,25 @@ static bool layered_monitor(void)
 	return true;
 }
 
+static bool timer_antistall_reset(void){
+	if(dsq_timer_check_flag) {
+		// flag was not cleared, ignore this tick.
+		return true;
+	} else {
+		// flag was cleared, reset it.
+		__sync_fetch_and_or(&dsq_timer_check_flag, 1);
+	}
+	return true;
+}
 
 static bool run_timer_cb(int key)
 {
 	switch (key) {
 	case LAYERED_MONITOR:
 		return layered_monitor();
 	case NOOP_TIMER:
+	case HACK_DSQ_CHECK:
+		return timer_antistall_reset();
```

Comment:
As this can put multiple CPUs into the antistall code path, it'd be better if the timer itself can scan and schedule antistall only when it's necessary. Also, note that, in newer kernels, timer can dispatch tasks directly to each CPU's local DSQs or the HI_FALLBACK. Given that we need to deploy to older kernels too, the current implementation is better for now.